### ISSUE TITLE
[apex] Support additional pattern for final properties

### ISSUE THREAD

**Issue by przemekrcloudity** [pmd/pmd#4758]
**Is your feature request related to a problem? Please describe.**

To make lazy loaded static constant (or final field) in Apex to avoid loading values that will not be needed in some cases on class loading context (when Salesforce loads class), we can make final properties in class, for example

```apex
public static class SomeClassName {
   public static final String HEAVY_CONSTANT {
        get {
            if (HEAVY_CONSTANT == null) {
                HEAVY_CONSTANT = someHeavyOperationForFetchingConstantValueForExampleFromDbSchemaDefinition();
            }
            return HEAVY_CONSTANT;
        }
        private set;
    }  
}
```

Unfortunately, this structure is not detected as constant by [PropertyNamingConventions](https://docs.pmd-code.org/latest/pmd_rules_apex_codestyle.html#propertynamingconventions) and this is technically a constant as its value can't be changed after initiazliation (verified in runtime).
The only difference from  constants supported by PMD is that their value is loaded lazily. 

**Describe the solution you'd like**

I would like [PropertyNamingConventions](https://docs.pmd-code.org/latest/pmd_rules_apex_codestyle.html#propertynamingconventions) to support lazily loaded constants (properties with final keyword) in same way as [FieldNamingConventions](https://docs.pmd-code.org/latest/pmd_rules_apex_codestyle.html#fieldnamingconventions).

**Describe alternatives you've considered**

Changing config of mentioned rule for static property pattern.
This would not work because some of static properties are not final and should not be treated as constants.

Another alternative was to suppress warnings for all lazily loaded constants.
This as you imagine makes code less readable.

**Additional context**
Add any other context about the feature request here.

Tested using command
```sh
pmd7-4rc check -d pathToFile -R category/apex/codestyle.xml/PropertyNamingConventions
```

Where `pmd7-4rc` is alias for path to PMD binary in the newest version available (7.0.0-rc4).
Class provided in issue description triggers this rule.
